File: common-account/src/main/java/tech/corefinance/account/common/model/DepositAccountInterestRate.java
package tech.corefinance.account.common.model;

import lombok.Data;
import tech.corefinance.product.common.enums.InterestCalculationMethod;
import tech.corefinance.product.common.enums.InterestDayInYear;
import tech.corefinance.product.common.model.*;

import java.util.List;

@Data
public class DepositAccountInterestRate {
    private DepositInterestRateTerms interestRateTerms;
    private InterestCalculationMethod interestCalculationMethod;
    private int percentPerDay;
    private DepositBalanceInterestCalculation balanceInterestCalculation;
    private InterestCalculationDateOptionType calculationDateType;
    private int calculationDateFixedMonth;
    private int calculationDateFixedDay;
    private InterestCalculationDateOptionType calculationDateOption;
    private InterestDayInYear interestDayInYear;

    private Boolean applyWithholdingTaxes;
    // Fixed interest rate
    private Boolean allowNegativeInterestRate;
    /**
     * Interest Rate Constraints (%) for fixed interest rate. <br/>
     * Interest Spread Constraints (%) for index rate source.
     */
    private List<CurrencyValue> interestRateValues;
    private String interestRateIndexSource;
    // Tiered interest rate
    private List<TieredInterestItem> interestItems;
}

File: adminweb/src/app/classes/accounts/DepositAccountInterestRate.ts
import { DepositBalanceInterestCalculation, DepositInterestRateTerms, InterestCalculationDateOptionType } from "../products/DepositProduct";
import { InterestCalculationMethod } from "../products/InterestCalculationMethod";
import { InterestDayInYear } from "../products/InterestDayInYear";
import { TieredInterestItem } from "../products/TieredInterestItem";
import { CurrencyLimitValue } from "../products/ValueConstraint";

export class DepositAccountInterestRate {
    interestRateTerms = DepositInterestRateTerms.FIXED;
    interestCalculationMethod: InterestCalculationMethod | null = InterestCalculationMethod.PERCENTAGE_PER_MONTH;
    percentPerDay = 0;
    balanceInterestCalculation: DepositBalanceInterestCalculation | null = DepositBalanceInterestCalculation.END_OF_dAY;
    calculationDateType: InterestCalculationDateOptionType = InterestCalculationDateOptionType.FIRST_DAY_EVERY_MOTNH;
    calculationDateFixedMonth = new Date().getMonth() + 1;
    calculationDateFixedDay = new Date().getDate();
    interestDayInYear: InterestDayInYear | null = InterestDayInYear.FIXED_365_DAYS;

    applyWithholdingTaxes: boolean | null = false;
    // Fixed interest rate
    allowNegativeInterestRate: boolean | null = false;
    /**
     * Interest Rate Constraints (%) for fixed interest rate. <br/>
     * Interest Spread Constraints (%) for index rate source.
     */
    interestRateValues: CurrencyLimitValue[] = [];
    sameInterestForAllCurrency = true;
    interestRateIndexSource: string = "";
    // Tiered interest rate
    interestItems: TieredInterestItem[] = [];
}